src/main/java/fr/cnes/sonar/plugins/icode/check/ICodeSensor.java

```diff
@@ -97,6 +100,33 @@ public void execute(SensorContext sensorContext) {
         final Configuration config = sensorContext.config();
         // Represent the active rules used for the analysis.
         final ActiveRules activeRules = sensorContext.activeRules();
+        // run i-Code CNES execution
+        if(config.getBoolean(ICodePluginProperties.AUTOLAUNCH_PROP_KEY).orElse(Boolean.getBoolean(ICodePluginProperties.AUTOLAUNCH_PROP_DEFAULT))) {
+            LOGGER.info("i-Code CNES auto-launch enabled.");
+            final String executable = config.get(ICodePluginProperties.ICODE_PATH_KEY).orElse(ICodePluginProperties.ICODE_PATH_DEFAULT);
+            final String[] files = sensorContext.fileSystem().baseDir().list();
+            final String outputFile = config.get(ICodePluginProperties.REPORT_PATH_KEY).orElse(ICodePluginProperties.REPORT_PATH_DEFAULT);
+            final String outputPath = Paths.get(sensorContext.fileSystem().baseDir().toString(),outputFile).toString();
+            final String outputOption = "-o";
+            final String command = String.join(" ", executable, String.join(" ",files), outputOption, outputPath);
+
+            LOGGER.info("Running i-Code CNES and generating results to "+ outputPath);
+            try {
+                final Process icode =  Runtime.getRuntime().exec(command);
+                int success = icode.waitFor();
+                if(0!=success){
+                    final String message = String.format("i-Code CNES auto-launch analysis failed with exit code %d.",success);
+                    throw new ICodeException(message);
+                }
+                LOGGER.info("Auto-launch successfully executed i-Code CNES.");
+            } catch (InterruptedException | IOException | ICodeException e) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Either re-interrupt this method or rethrow the "InterruptedException". ([squid:S2142](https://sonarcloud.io/organizations/lequal/rules#rule_key=squid%3AS2142))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fr.cnes.sonarqube.plugins%3Asonaricode&pullRequest=33&issues=AWXXG66W_eEZPtUm_0Ak&open=AWXXG66W_eEZPtUm_0Ak)

[ISSUE_KEY]: AWXXG66W_eEZPtUm_0Ak